tough/src/editor/targets.rs

```diff
@@ -0,0 +1,556 @@
+    pub fn from_targets(name: &str, targets: Targets, key_holder: KeyHolder) -> Self {
+        TargetsEditor {
+            key_holder: Some(key_holder),
+            delegations: targets.delegations,
+            new_targets: None,
+            existing_targets: Some(targets.targets),
+            version: None,
+            expires: None,
+            name: name.to_string(),
+            new_roles: None,
+            _extra: Some(targets._extra),
+            limits: None,
+            transport: None,
+        }
+    }
+
+    /// Creates a `TargetsEditor` with the provided targets from an already loaded repo
+    /// `version` and `expires` are thrown out to encourage updating the version and expiration
+    /// If a `Repository` has been loaded, use `from_repo()` to preserve the `Transport` and `Limits`.
+    pub fn from_repo(repo: &Repository<'a, T>, name: &str) -> Result<Self>
+    where
+        T: Transport,
+    {
+        let targets = repo
+            .delegated_role(name)
+            .context(error::DelegateNotFound {
+                name: name.to_string(),
+            })?
+            .targets
+            .as_ref()
+            .context(error::NoTargets)?
+            .signed
+            .clone();
+        let key_holder = KeyHolder::Delegations(
+            repo.targets
+                .signed
+                .parent_of(name)
+                .context(error::DelegateMissing {
+                    name: name.to_string(),
+                })?
+                .clone(),
+        );
+        Ok(TargetsEditor::<'a, T> {
+            key_holder: Some(key_holder),
+            delegations: targets.delegations,
+            new_targets: None,
+            existing_targets: Some(targets.targets),
+            version: None,
+            expires: None,
+            name: name.to_string(),
+            new_roles: None,
+            _extra: Some(targets._extra),
+            limits: Some(repo.limits),
+            transport: Some(repo.transport),
+        })
+    }
+
+    /// Adds limits to the `TargetsEditor`, only necessary if loading a role
+    pub fn limits(&mut self, limits: Limits) {
+        self.limits = Some(limits);
+    }
+
+    /// Add a transport to the `TargetsEditor`, only necessary if loading a role
+    pub fn transport(&mut self, transport: &'a T) {
+        self.transport = Some(transport);
+    }
+
+    /// Add a `Target` to the `Targets` role
+    pub fn add_target(&mut self, name: &str, target: Target) -> &mut Self {
+        self.new_targets
+            .get_or_insert_with(HashMap::new)
+            .insert(name.to_string(), target);
+        self
+    }
+
+    /// Add a target to the repository using its path
+    ///
+    /// Note: This function builds a `Target` synchronously;
+    /// no multithreading or parallelism is used. If you have a large number
+    /// of targets to add, and require advanced performance, you may want to
+    /// construct `Target`s directly in parallel and use `add_target()`.
```

Comment:
Seems like I could also use this function to roll my own parallelization. Seems like this warning is only relevant for a function that operates on a batch, or directory walk.